### DISCUSSION TITLE
Disable OS specific defines

### DISCUSSION THREAD

**Discussion by mrjbom** [microsoft/vscode-cpptools#8625] [Q&A] [answered]
I'm using a cross-compiler for my project and I'm having problems with how IntelliSense interprets the ```__linux__``` define

For this code
```
#ifdef __linux__
#error "This is not intended to be compiled for Linux"
#endif
```
IntelliSense show that ```__linux__``` is defined to 1 and i get error in VS Code
However, when I use my cross-compiler, I do not get errors, because ```__linux__``` is not declared there
I would like to make it so that IntelliSense also doesn't see ```__linux__``` define

Here are my IntelliSense settings
```
{
        "name": "i686",
        "includePath": [
            "${default}",
            "/home/username/opt/cross/lib/gcc/i686-elf/11.2.0/include"
        ],
        "defines": [],
        "cStandard": "gnu99",
        "cppStandard": "gnu++14",
        "intelliSenseMode": "linux-gcc-x86",
        "compilerPath": "/home/username/opt/cross/bin/i686-elf-gcc",
        "compilerArgs": [
            "-ffreestanding"
        ]
}
```

I noticed that ```intelliSenseMode``` affects this, because if I put something unrelated to Linux there, then everything is fine, but I do not know how to choose just ```x86-gcc``` there, without binding to my or another platform.

**Comment by sean-mcmanus** [comment, marked as answer]
This is related to https://github.com/microsoft/vscode-cpptools/issues/4653 , not sure if we need a separate issue for gcc on a non-Linux OS.

**Reply by sean-mcmanus** [reply]
Our extension will define it internally when a linux intelliSenseMode is used.
